### DISCUSSION TITLE
It is not able to use the route filter.

### DISCUSSION THREAD

**Discussion by Ferks-FK** [tighten/ziggy#646] [Q&A]
### Ziggy version

1.6

### Laravel version

9.19

### Description

I am trying to filter the routes according to the route accessed according to this part of the [guide](https://github.com/tighten/ziggy#filtering-routes).

I'm trying to make use of the groups, but it's not working anyway.
I have 2 layouts, one for unauthenticated users (app) and one for admins (admin).
Here is my `config/ziggy.php`.

```php
'groups' => [
        'admin' => ['admin.*'],
        'app' => [
            'home.index',
            'auth.*',
            'servers.*',
            'bans.*',
            'mutes.*',
            'report.*',
            'appeal.*',
            'steam.*',
            'locales.*'
        ]
    ],
    'except' => [
        'debugbar.*',
        'ignition.*'
    ]
```

```php
// app.blade.php
@routes('app')

// admin.blade.php
@routes('admin')
```
All my routes are group-based.
No matter how much I use the `app` layout, all the administration routes are visible.

### Ziggy call and context

```js
In all my React components.
```

### Ziggy configuration

```js
I don't have any relevant settings.
I have just that in my HandleInertiaRequest Middleware:

'ziggy' => function () use ($request) {
  return array_merge((new Ziggy)->toArray(), [
     'location' => $request->url()
  ]);
},
```

### Route definition

```php
Route::name('admin.')->group(function() {
   Route::group(['prefix' => 'admin_settings'], function() {
        Route::get('/', [AdminController::class, 'index'])->name('settings.index');
        Route::get('/create', [AdminController::class, 'create'])->name('settings.create');
        Route::get('/{id}', [AdminController::class, 'show'])->name('settings.show');
        Route::post('/store', [AdminController::class, 'store'])->name('settings.store');
        Route::patch('/update/{id}', [AdminController::class, 'update'])->name('settings.update');
        Route::delete('/{id}', [AdminController::class, 'destroy'])->name('settings.destroy');
    });
    .....much more

}
```

**Comment by bakerkretzmar** [comment]
See https://github.com/tighten/ziggy/discussions/465 and https://github.com/tighten/ziggy/discussions/463#discussioncomment-1381011.

The `@routes()` Blade directive only runs when the view it's in is rendered, so with Inertia that'll happen on the first page load but not subsequent ones. You can verify that this is what's happening by hard-refreshing or navigating directly to an non-admin page, which should show the correct routes.

To get an updated route list on every page you'll have to reload the Ziggy config on every navigation event, or at least every navigation event between admin and non-admin pages. See the linked discussions for how to do that.

If your routes are all authenticated and authorized properly though, consider just not bothering hiding the admin routes. Having them appear in the HTML or request payload isn't a security risk.

Let me know if that works and if you have any other questions!
